File: frontend/src/api/sentiment.ts
import { apiRequest } from "./client";
import type { SentimentAnalyzeResponse } from "../features/intelligence/types";

const SENTIMENT_BASE_PATH = "/api/v1/sentiment/portfolios";

export function analyzePortfolioSentiment(portfolioId: number): Promise<SentimentAnalyzeResponse> {
  return apiRequest<SentimentAnalyzeResponse>(`${SENTIMENT_BASE_PATH}/${portfolioId}/analyze`, {
    method: "POST",
  });
}

File: backend/app/api/v1/endpoints/sentiment.py
"""
Sentiment analysis endpoints.
"""

from typing import Annotated

from fastapi import APIRouter, Path

from app.api.deps import CurrentUserDep, SentimentServiceDep
from app.schemas.sentiment import SentimentAnalyzeResponse

router = APIRouter()


@router.post(
    "/portfolios/{portfolio_id}/analyze",
    response_model=SentimentAnalyzeResponse,
)
async def analyze_portfolio_sentiment(
    portfolio_id: Annotated[int, Path(ge=1)],
    service: SentimentServiceDep,
    current_user: CurrentUserDep,
) -> SentimentAnalyzeResponse:
    """Analyze local portfolio news and return article/holding/portfolio sentiment."""
    return await service.analyze_portfolio_sentiment(portfolio_id, current_user.id)

